.github/workflows/publish-pacts.yaml

```diff
@@ -40,8 +43,17 @@ jobs:
       contents: 'read'
       id-token: 'write'
     steps:
+      - name: Bump the tag to a new version
```

Comment:
I don't think this belongs in a "publish-pacts" workflow. Also, we already have a `tag-publish` workflow that does this and pushes the new version to Sherlock. Sounds like we may need a wider refactor of workflows... maybe a release workflow that can then call other workflows?

Reply:
Notice this tag uses the `DRY_RUN` feature. The `tag-publish` workflow does not. What this means is that the `publish-pacts.yml` workflow will look ahead to get the next tag BEFORE the tag is actually dispatched to the repository via `tag-publish`. This will maintain versioning consistency between the published pact and the release tag. Hope this makes sense.

Reply:
👍  Alright, I'll accept that I don't know what's going on here and leave this PR with a request for detailed comments. The web of workflows connected by `workflow_dispatch` / `workflow_call` is difficult to understand and it has been my observation that Pact stuff especially is well understood by few developers.